app/components/MyComponent.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  Text,
+  View,
+  TouchableOpacity,
+  FlatList,
+  TextInput,
+} from "react-native";
+
+const MyComponent = ({ data }) => {
+  const [selectedItems, setSelectedItems] = useState([]);
+  const [dataSource, setDataSource] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    setDataSource(data);
+  }, [data]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setDataSource(data.filter((item) => item.name.includes(searchTerm)));
+    }, 1000);
+  }, [searchTerm]);
+
+  const handleSelect = (item) => {
+    setSelectedItems((currentSelectedItems) => [...currentSelectedItems, item]);
+  };
```

Comment:
This is a buggy! 

propose of this implementation is:
`Each item can be selected or not selected, and this state is toggled by tapping on the item` 

current implementation add item to array so we have to add functionality to remove items if previously selected.


```suggestion
  const handleSelect = (item: DataType) => {
    const filteredItems = selectedItems.filter((v) => v.id != item.id);
    if (filteredItems.length == selectedItems.length) {
      setSelectedItems((currentSelectedItems) => [
        ...currentSelectedItems,
        item,
      ]);
    } else {
      setSelectedItems(filteredItems);
    }
  };
```